paddle/gserver/gradientmachines/RecurrentGradientMachine.cpp

```diff
@@ -429,7 +430,11 @@ void RecurrentGradientMachine::reorganizeInput(PassType passType) {
   }
 
   {
-    AsyncGpuBlock asyncGpuBlock;
+    std::unique_ptr<AsyncGpuBlock> asyncBlock;
```

Comment:
seems no other place use this Block? ? ?.
I am really confusing about Paddle's init logic, almost a black box to me.

Reply:
Well, that is very amazing C++ technic which called `RAII`.

It just creates some resource when the `constructor` is invoked, and release them when the `deconstructor` is invoked. It is something similar with `with` keyword in `Python`.

It is widely used in `std` library, such as `unique_ptr`, `lock_guard`, `shared_ptr`, etc.

More details, please refer that [wiki page](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization)